pkg/controllers/status/common.go

```diff
@@ -144,8 +144,10 @@ func updateResourceStatus(
 		newObj, err := interpreter.AggregateStatus(resource, bindingStatus.AggregatedStatus)
 		if err != nil {
 			klog.Errorf("Failed to aggregate status for resource template(%s/%s/%s), Error: %v", gvr, resource.GetNamespace(), resource.GetName(), err)
+			eventRecorder.Event(resource, corev1.EventTypeWarning, events.EventReasonAggregateStatusFailed, err.Error())
 			return err
 		}
+		eventRecorder.Event(resource, corev1.EventTypeNormal, events.EventReasonAggregateStatusSucceed, "Aggregated status to object successfully.")
```

Comment:
Why do you need to adjust the position of these two events?

Reply:
I tend to add events after each resource interpretation function to reflect whether the resource interpretation is successful.

Reply:
+1
but I don't have a suitable name to distinguish, do you have one?

Reply:
thanks
@RainbowMango WDYT?

Reply:
> I tend to add events after each resource interpretation function to reflect whether the resource interpretation is successful.

Can you remind me of the benefit we can get from it?

Reply:
we can more easily view the runtime of the resource interpretation framework via k8s events, and this can also be connected to the alarm center to notify the administrator or developer.

Reply:
So you want the `AggregateStatusSucceed` and `AggregateStatusFailed` dedicated to indicating resource interpreter status? That means that updating failure after `interpreter.AggregateStatus`, should not report event.

Reply:
yes, such update error can be fixed by reconciliation.

Reply:
ping @RainbowMango

Reply:
But at the time emitting the success event, the aggregate operation might still not happen, in addition, there might be more than one events due to the retry loop. 

I'm afraid it is not acceptable.

Reply:
other functions of the current resource interpretation framework are also designed in this way, so another purpose of this PR is to be consistent with them.

* `ReflectStatus`:
https://github.com/karmada-io/karmada/blob/836b8789d7e679fe579bc02d2ab2fb994435c2c2/pkg/controllers/status/work_status_controller.go#L359-L366

* `InterpretHealth`:
https://github.com/karmada-io/karmada/blob/836b8789d7e679fe579bc02d2ab2fb994435c2c2/pkg/controllers/status/work_status_controller.go#L371-L380

* `GetDependencies`:
https://github.com/karmada-io/karmada/blob/836b8789d7e679fe579bc02d2ab2fb994435c2c2/pkg/dependenciesdistributor/dependencies_distributor.go#L271-L277